runtime/moonbase/tests/xcm_tests.rs

```diff
@@ -922,6 +924,285 @@ fn transact_through_sovereign() {
+	MockNet::reset();
+
+	let para_a_balances = MultiLocation::new(1, X2(Parachain(1), PalletInstance(1u8)));
+	let source_location = parachain::AssetType::Xcm(para_a_balances);
+	let source_id: parachain::AssetId = source_location.clone().into();
+
+	let asset_metadata = parachain::AssetMetadata {
+		name: b"ParaAToken".to_vec(),
+		symbol: b"ParaA".to_vec(),
+		decimals: 18,
+	};
+	let response = Response::Version(2);
+
+	// This is irrelevant, nothing will be done with this message,
+	// but we need to pass a message as an argument to trigger the storage change
+	let mock_message: Xcm<()> = Xcm(vec![QueryResponse {
+		query_id: 0,
+		response,
+		max_weight: 0,
+	}]);
+
+	ParaA::execute_with(|| {
+		// advertised version
+		parachain::XcmVersioner::set_version(2);
+	});
+
+	ParaB::execute_with(|| {
+		// Let's try with v0
+		parachain::XcmVersioner::set_version(0);
+
+		assert_ok!(AssetManager::register_asset(
+			parachain::Origin::root(),
+			source_location,
+			asset_metadata,
+			1u128,
+		));
+		assert_ok!(AssetManager::set_asset_units_per_second(
+			parachain::Origin::root(),
+			source_id,
+			0u128
+		));
+	});
+
+	ParaA::execute_with(|| {
+		// This sets the default version, for not known destinations
+		assert_ok!(ParachainPalletXcm::force_default_xcm_version(
+			parachain::Origin::root(),
+			Some(2)
+		));
+		// Wrap version, which sets VersionedStorage
+		assert_ok!(<ParachainPalletXcm as WrapVersion>::wrap_version(
+			&MultiLocation::new(1, X1(Parachain(2))).into(),
+			mock_message
+		));
+
+		parachain::para_roll_to(2);
+
+		// queries should have been updated
+		assert!(ParachainPalletXcm::query(0).is_some());
+	});
+
+	let expected_supported_version: parachain::Event = pallet_xcm::Event::SupportedVersionChanged(
+		MultiLocation {
+			parents: 1,
+			interior: X1(Parachain(2)),
+		},
+		0,
+	)
+	.into();
+
+	ParaA::execute_with(|| {
+		// Assert that the events vector contains the version change
+		assert!(parachain::para_events().contains(&expected_supported_version));
+	});
+
+	// Let's ensure talking in v0 works
+	let dest = MultiLocation {
+		parents: 1,
+		interior: X2(
+			Parachain(2),
+			AccountKey20 {
+				network: NetworkId::Any,
+				key: PARAALICE.into(),
+			},
+		),
+	};
+	ParaA::execute_with(|| {
+		// free execution, full amount received
+		assert_ok!(XTokens::transfer(
+			parachain::Origin::signed(PARAALICE.into()),
+			parachain::CurrencyId::SelfReserve,
+			100,
+			Box::new(dest),
+			80
+		));
+		// free execution, full amount received
+		assert_eq!(
+			ParaBalances::free_balance(&PARAALICE.into()),
+			INITIAL_BALANCE - 100
+		);
+	});
+
+	ParaB::execute_with(|| {
+		// free execution, full amount received
+		assert_eq!(Assets::balance(source_id, &PARAALICE.into()), 100);
+	});
+
+	let expected_version_notified: parachain::Event = pallet_xcm::Event::VersionChangeNotified(
+		MultiLocation {
+			parents: 1,
+			interior: X1(Parachain(1)),
+		},
+		2,
```

Comment:
~~These tests have a lot of code in common, and it's hard to pick out what differs about them and why. Here where you define the expectation (or somewhere else in the test), it might be useful to add a brief comment about what it is that you're testing~~

Lol, you literally do that 4 lines below...